src/core/components.tsx

```diff
@@ -1305,9 +1305,12 @@ export const MultiSelectMenu = (props: MultiSelectMenuProps) => (
       const [{ fields: optionsOrGroups }, optionPromises] = reconcile(
         React.createElement(Section, { children: props.children })
       );
-      const [{ fields: initialOptions }, initialOptionsPromises] = reconcile(
+      const [, initialOptionsPromises] = reconcile(
```

Comment:
For some reason, I had to keep calling reconcile on `React.createElement(Section, ...`, otherwise the initial options wouldn't render